### ISSUE TITLE
About gt distance using faiss

### ISSUE THREAD

**Issue by HospitableHost** [garvita-tiwari/PoseNDF#18]
Hi, I have tried to get gt distance of AMASS‘s plausible poses using faiss, but found that some plausible poses' gt distance were not 0. Obviously, they should be. For example, for some poses from SSM_synced, the gt distance using faiss is shown in the screenshot below.
![image](https://user-images.githubusercontent.com/44576391/218918122-b1c6933a-dd5e-493d-9fcf-6311424b8435.png)
So, is it necessary to use faiss search for AMASS‘s ’plausible poses? Can I directly assign gt distance of  AMASS‘s ’plausible poses to 0?

**Comment by garvita-tiwari** [comment]
For AMASS you should directly use 0 as the distance.
For a sample pose for SSM_synced distance is not 0, because SSM_synced is not part of training set.

Are you using the complete code to get this distance or this is just with faiss?? 
Note that we use faiss to get approximate distance and then we calculate geodesic distance based on the 500 nearest neighbour we get using faiss.

**Comment by HospitableHost** [comment]
> For AMASS you should directly use 0 as the distance. For a sample pose for SSM_synced distance is not 0, because SSM_synced is not part of training set.
> 
> Are you using the complete code to get this distance or this is just with faiss?? Note that we use faiss to get approximate distance and then we calculate geodesic distance based on the 500 nearest neighbour we get using faiss.

Hi, thank you for your reply!
I got your point. The training set is the zero level. Though the poses from SSM_synced are all plausible, their distance also should be calculated by faiss and knn. But the distance result seems unreasonable.
I use the complete code to get this distance(including faiss, knn and geodesic distance).

**Comment by garvita-tiwari** [comment]
Hi, Yes the poses from SSM_synced are plausible, but since they are not part of the training data, the distance will not be zero. Instead you should get some non-zero small distance.
Can you share the exact poses, you are using here?

**Comment by HospitableHost** [comment]
![image](https://user-images.githubusercontent.com/44576391/220294755-a26af061-eaba-4a63-9579-21dcecb6c441.png)
![image](https://user-images.githubusercontent.com/44576391/220293716-332eda96-045c-4cd4-b73a-a534be71564f.png)
Hey, I draw two figures to analyze data. It shows that all npz files from vald is not reasonable. Besides several npz files from test are also not reasonable.

**Comment by garvita-tiwari** [comment]
Hi, thanks for sharing this plot. I have some questions
1. Are you using AMASS training for finding distance?
2.  What is x-axis in the plot ?

**Comment by HospitableHost** [comment]
> Hi, thanks for sharing this plot. I have some questions
> 
> 1. Are you using AMASS training for finding distance?
> 2. What is x-axis in the plot ?

Yes, I only use AMASS training set for finging the distance.
The x-asis refers to npz file name, which has no specific meaning.

**Comment by HospitableHost** [comment]
> Hi, Yes the poses from SSM_synced are plausible, but since they are not part of the training data, the distance will not be zero. Instead you should get some non-zero small distance. Can you share the exact poses, you are using here?

Hi, I fixed this bug. It's caused by a bug in sample_poses.py, which is showed in the screenshot below.
![image](https://user-images.githubusercontent.com/44576391/221342553-e4c68318-0fa6-4851-9504-1bdc2e7dbe93.png)
The original version of the code will lead to poses redundancy. As a result, the data is redundant from 1.2GB to 33GB. And thus, faiss will not be so accurate when performing similarity searches

**Comment by garvita-tiwari** [comment]
Hi, thanks for noticing this. We had already corrected this here : 
https://github.com/garvita-tiwari/PoseNDF/commit/1dc1466bad979a39759de83f56f8dfd231307a4f#diff-bd55344acf61c76e853ff870efd9eb190466da0913cc3ee52bfff911ed78a1d4
